students/Yue_Ma/Lesson03/mailroom.py

```diff
@@ -54,32 +50,33 @@ def send_a_thank_you_note():
                 donor_list[user_name] = list_history
                 print('Updated the info to our database.')
                 print(email(user_name))
-                back_to_main()
             else:
                 print('')
                 print('This name is NOT in our database!')
                 donor_list[user_name] = [float(user_amount)]
                 print('Added the name to our database.')
                 print(email(user_name))
-                back_to_main()
-            break
         elif user_send_a_thank_you_note == '2':
-            back_to_main()
             break
         else:
             print("Not a valid option!!!")
+    return main()
```

Comment:
you don't want to call back to the main here as well, just `return` is sufficient